src/dune_pkg/opam.ml

```diff
@@ -315,6 +326,13 @@ module Repo_selection = struct
       ; env
       }
 
+  let with_sys_env sys_env = function
+    | Switch_with_name _ as rs -> rs
```

Comment:
No, the problem is that in a `Switch_context` it will [always call OPAM](https://github.com/ocaml-opam/opam-0install-solver/blob/a17703ba83374975633d66bea2d40da51beabee5/lib/switch_context.ml#L22) so it is not possible to inject custom variables in a `Switch_context`. In `Dir_context` it will call a [custom function](https://github.com/ocaml-opam/opam-0install-solver/blob/a17703ba83374975633d66bea2d40da51beabee5/lib/dir_context.ml#L66).

So we could 

* do nothing
* print a warning that that this is missing functionality in this branch
* submit a PR upstream to closer align the `Dir_context` and `Switch_context`

Reply:
For now we decided to not do anything. As we have learned in an in-person discussion with the OPAM maintainers, the opam-repository is not actually stored in an unpacked format by any stable version of OPAM at the moment, so we can't use `Dir_context` and point it at the unpacked opam-repository of the switch. Given this support is temporary until we implement our own opam-repository fetching this is probably good enough for now.